frontend/js/src/user/stats/components/UserArtistMap.tsx

```diff
@@ -126,99 +73,98 @@ export default class UserArtistMap extends React.Component<
     });
   };
 
-  render() {
-    const {
-      selectedMetric,
-      data,
-      errorMessage,
-      hasError,
-      loading,
-    } = this.state;
+  const [selectedMetric, setSelectedMetric] = React.useState<
+    "artist" | "listen"
+  >("artist");
 
-    return (
-      <Card className="user-stats-card">
-        <div className="row">
-          <div className="col-md-9 col-xs-6">
-            <h3 style={{ marginLeft: 20 }}>
-              <span className="capitalize-bold">Artist Origins</span>
-              <small>&nbsp;Click on a country to see more details</small>
-            </h3>
-          </div>
-          <div
-            className="col-md-2 col-xs-4 text-right"
-            style={{ marginTop: 20 }}
-          >
-            <span>Rank by</span>
-            <span className="dropdown">
-              <button
-                className="dropdown-toggle btn-transparent capitalize-bold"
-                data-toggle="dropdown"
-                type="button"
+  const [data, setData] = React.useState<UserArtistMapData>([]);
+
+  React.useEffect(() => {
+    if (rawData && rawData?.payload) {
+      const newProcessedData = processData(selectedMetric, rawData);
+      setData(newProcessedData);
+    }
+  }, [rawData, selectedMetric]);
+
+  const changeSelectedMetric = (newSelectedMetric: "artist" | "listen") => {
+    setSelectedMetric(newSelectedMetric);
+  };
```

Comment:
You can delete this function and use `setSelectedMetric` directly in the pill buttons below